recipes/gdbm/all/conanfile.py

```diff
@@ -96,7 +97,13 @@ def package(self):
             autotools = self._configure_autotools()
             autotools.install()
         os.unlink(os.path.join(self.package_folder, "lib", "libgdbm.la"))
+        compat_la = os.path.join(self.package_folder, "lib",
+                                 "libgdbm_compat.la")
+        if (os.path.exists(compat_la)):
+            os.unlink(compat_la)
         tools.rmdir(os.path.join(self.package_folder, "share"))
 
     def package_info(self):
-        self.cpp_info.libs = ["gdbm"]
+        if self.options.libgdbm_compat:
+            self.cpp_info.libs.append("gdbm_compat")
+        self.cpp_info.libs.append("gdbm")
```

Comment:
gettext is only needed for the executable, not the libraries.
So when adding `gdbm` as a requirement, how can I avoid adding `gettext` as a requirement by default.

Does this work? Are there simpler ways?
@jgsogo @danimtb 

```suggestion
        if self.options.libgdbm_compat:
            self.cpp_info.components["ALL"].libs.append("gdbm_compat")
        self.cpp_info.components["ALL"].libs.append("gdbm")
        if self.options.with_libiconv:
            self.cpp_info.comonents["ALL"].requires.append("libiconv::libiconv")
        if self.options.with_gettext:
            self.cpp_info.comonents["_gettext"].requires.append("gettext::gettext")
```

Reply:
(I think this is speculative, there is no such `with_gettext` option or requirement in this recipe)

I don't know why you are proposing here an `ALL` component, when all these `libs` and `requires` are handled by some other option:
 * `gdbm_compat` is not generated neither collected if the option `libgdbm_compat` is deactivated.
 * the option `with_gettext` would handle the requirement for `gettext` and it will be added or not in the `configure()` method depending on the value of that option.

Reply:
Once again, I confused `gettext` with `readline`.

See http://git.gnu.org.ua/cgit/gdbm.git/tree/src/Makefile.am#n91,
Here you see that the readline is only added to the programs, not to the library (at http://git.gnu.org.ua/cgit/gdbm.git/tree/src/Makefile.am#n41)

Reply:
Yes, it will work as you expect if the consumer uses components too and make it explicit the components from `gdbm` it is using:

**gdbm recipe**
```python
def requirements(self):
    self.requires("gettext/version")

def package_info(self):
    self.cpp_info.components["gdbmlib"].libs = ["gdbm"]
    self.cpp_info.components["gdbmtool"].requires = ["gdbm", "gettext::gettext"]
```

**consumer recipe**
```python
def requirements(self):
     self.requires("gdbm/version")

def package_info(self):
    self.cpp_info.components["libA"].requires = ["gdbm::gdbmlib"]  # Only the library
```

@danimtb took some nice pictures of the graph dependencies generated by CMake: https://github.com/conan-io/conan/pull/7108#issuecomment-636711636

Take into account that components only work for the `cmake_find_package[_multi]` generator, consuming a library with components using any other generator will consume use it as a single unit.

Reply:
Indeed, this is for components only.

So the `readline` requirement should be tagged `private`?

Reply:
I'm not sure. If everything was working without any bug it could be a solution. You'd hide `readline` from the consumers, but still the runtime information is needed when linked dynamically and executed down the graph. Also, hiding it from consumers will let you use different versions of the same `readline` requirement without raising a conflict. This would be great, and it is how it should work, but I'm afraid there are some issues related to `privates` and better to avoid them until we have an improved graph model. I need to try and experiment this very specific use case before recommending `privates` in this situation.

Reply:
> but still the runtime information is needed when linked dynamically and executed down the graph

Only the location of the libraries should be added to `LD_LIBRARY_PATH`/`DYLD_LIBRARY_PATH`/`PATH` on Linux/Mac/Windows.

Reply:
Yes, but right now there is no such distinction. If runtime information is propagated, everything will. I started to think about these issues in this repo (https://github.com/jgsogo/conan-poc-graph-cpp), it is a draft and it was written for myself, so probably it is hard to understand. We really think we need a better and improved graph model for the next major and we are working on it trying to gather the best feedback asap (that's also the reason why there are so many comments about the _host_ and _build_ contexts and the two profiles)

Reply:
Is there an action item left here?

Reply:
I think we can move forward the PR and add components later